functions/Update-DbaLocalCachedSoftware.ps1

```diff
@@ -0,0 +1,210 @@
+function Update-DbaLocalCachedSoftware {
+    <#
+    .SYNOPSIS
+        Download and extract software from Github to update the local cached version of that software.
+
+    .DESCRIPTION
+        Download and extract software from Github to update the local cached version of that software.
+        This command is run from inside of Install-Dba* and Update-Dba* commands to update the local cache if needed.
+
+    .PARAMETER Software
+        Name of the software to download.
+        Options include:
+        * MaintenanceSolution: SQL Server Maintenance Solution created by Ola Hallengren (https://ola.hallengren.com)
+        * FirstResponderKit: First Responder Kit created by Brent Ozar (http://FirstResponderKit.org)
+        * DarlingData: Erik Darling's stored procedures (https://www.erikdarlingdata.com)
+
+    .PARAMETER Branch
+        Specifies the branch. Defaults to master or main. Can only be used if Software is used.
+
+    .PARAMETER LocalFile
+        Specifies the path to a local file to install from instead of downloading from Github.
+
+    .PARAMETER Url
+        Specifies the URL to download from. Is not needed if Software is used.
+
+    .PARAMETER LocalDirectory
+        Specifies the local directory to extract the downloaded file to. Is not needed if Software is used.
+
+    .PARAMETER WhatIf
+        If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+
+    .PARAMETER Confirm
+        If this switch is enabled, you will be prompted for confirmation before executing any operations that change state.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Community
+        Author: Andreas Jordan, @JordanOrdix
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+         https://dbatools.io/Update-DbaLocalCachedSoftware
+
+    .EXAMPLE
+        PS C:\> Update-DbaLocalCachedSoftware -Software MaintenanceSolution
+
+        Updates the local cache of Ola Hallengren's Solution objects.
+
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Medium")]
+    param (
+        [ValidateSet('MaintenanceSolution', 'FirstResponderKit', 'DarlingData')]
```

Comment:
Is this Mandatory? or can it do all at once? I couldn't quite figure it out by glancing at the code.

Reply:
No, not mandatory. You have two options: Using -Software to update that software with default settings. This is used by Update-DbaMaintananceSolution: `Update-DbaLocalCachedSoftware -Software MaintenanceSolution -LocalFile $LocalFile -EnableException`.
Or you could use the command to download any other software from Github by using -Url and -LocalDirectory. Don't know if this would be used very often, but it's only a few extra lines for beeing flexible.
And I decided to not introduce a loop to update more than one software, because our commands only need their software. Only a user might want to update more that one software maybe once a week or so - but then the user has to use the command more than one time or build a loop.